weather_report/utils.py

```diff
@@ -0,0 +1,24 @@
+"""utils.py contains utils functions for main.py
+"""
+
+
+def year_boundary_check(year: int) -> bool:
+    """
+    Return true if year is between 2004 and 2016
+    and false otherwise
+    """
+    if year and 2004 <= year <= 2016:
+        return True
+    return False
```

Comment:
You can provide this function in `type` argument of `add_argument` function. This will store only valid year values otherwise will give error.